From d12e4bc3de2bbd4d2e469154005d1bcdfd5be2e0 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Wed, 10 Jul 2024 17:00:52 +0300
Subject: [PATCH] [FSW-2704] fs1: wait for dma finish of image

This patch adds functionality to wait for current dma
image to finish. If manifest for this image is invalid
then it can return issuing next dma chunk.

Change-Id: I423b548c0c7013b28e348a2160b09ff986dd0be8
---

diff --git a/app/boot_rom.c b/app/boot_rom.c
index 9926a0e..f5dcd9a 100644
--- a/app/boot_rom.c
+++ b/app/boot_rom.c
@@ -61,6 +61,9 @@
 
 	boot_rom_boot_flow_validate_manifests();
 
+	/* Wait for ppboot spi dma to complete */
+	boot_rom_boot_flow_wait_for_dma_finish();
+
 	/* If both manifests are invalid enter into recovery */
 	if (!boot_rom_boot_flow_is_manifest_valid(HL_FW_SPI_MANIFEST_0) &&
 		!boot_rom_boot_flow_is_manifest_valid(HL_FW_SPI_MANIFEST_1)) {
diff --git a/common/boot_flow.c b/common/boot_flow.c
index 0670dd0..2844f97 100644
--- a/common/boot_flow.c
+++ b/common/boot_flow.c
@@ -124,3 +124,15 @@
 	else
 		return secondary_manifest_valid;
 }
+
+void boot_rom_boot_flow_wait_for_dma_finish(void)
+{
+	/*
+	 * TODO: Check for existing dma call to finish
+	 * If current dma image manifest validation is pass then
+	 * Start dma for remaining chunks
+	 * Wait for entire dma to finish
+	 * otherwise start prefetch of next image from different manifest
+	 */
+
+}
diff --git a/common/boot_flow.h b/common/boot_flow.h
index 3968321..492a152 100644
--- a/common/boot_flow.h
+++ b/common/boot_flow.h
@@ -10,5 +10,6 @@
 void boot_rom_boot_flow_boot_ppboot(enum hl_fw_manifest_type man_idx);
 void boot_rom_boot_flow_validate_manifests(void);
 bool boot_rom_boot_flow_is_manifest_valid(enum hl_fw_manifest_type man_idx);
+void boot_rom_boot_flow_wait_for_dma_finish(void);
 
 #endif /* _BOOT_FLOW_H_ */
